src/kernel/sporadic.c

```diff
@@ -323,15 +332,11 @@ void refill_unblock_check(sched_context_t *sc)
         NODE_STATE(ksReprogram) = true;
 
         /* merge available replenishments */
-        while (!refill_single(sc)) {
+        while (refill_unblock_check_mergable(sc)) {
             ticks_t amount = REFILL_HEAD(sc).rAmount;
-            if (REFILL_INDEX(sc, refill_next(sc, sc->scRefillHead)).rTime <= NODE_STATE_ON_CORE(ksCurTime, sc->scCore) + amount) {
-                refill_pop_head(sc);
-                REFILL_HEAD(sc).rAmount += amount;
-                REFILL_HEAD(sc).rTime = NODE_STATE_ON_CORE(ksCurTime, sc->scCore);
-            } else {
-                break;
```

Comment:
huh. Should this be added to the verification style guide for the kernel?

Reply:
It probably should. I also need to check with verification about what kinds of expressions can be used in conditions.

Reply:
happy for this commit to go in assuming that the change is semantically equivalent to what it was before with the break